### PR TITLE
Align max_pool1d Error Checking between CPU and CUDA/CPU requires_grad

### LINKED ISSUE #85712: max_pool1d error messages diverge between CPU and CUDA
### 🐛 Describe the bug

Ref: https://github.com/pytorch/pytorch/blob/c8a55b3150988f3e1f632e3bc4a6e47623d3f7b5/torch/testing/_internal/common_methods_invocations.py#L3118

https://github.com/pytorch/pytorch/blob/c8a55b3150988f3e1f632e3bc4a6e47623d3f7b5/torch/testing/_internal/common_methods_invocations.py#L3125

https://github.com/pytorch/pytorch/blob/c8a55b3150988f3e1f632e3bc4a6e47623d3f7b5/torch/testing/_internal/common_methods_invocations.py#L3133

https://github.com/pytorch/pytorch/blob/c8a55b3150988f3e1f632e3bc4a6e47623d3f7b5/torch/testing/_internal/common_methods_invocations.py#L3141

https://github.com/pytorch/pytorch/blob/c8a55b3150988f3e1f632e3bc4a6e47623d3f7b5/torch/testing/_internal/common_methods_invocations.py#L3148

Surfaced in PR : #85594 

### Versions

master

### PR DESCRIPTION
**Pull request by deandyu**
Fixes https://github.com/pytorch/pytorch/issues/85712

Standardizes error checking for max_pool1d between CPU and CPU requires_grad/CUDA.

### CODE REVIEW

File: aten/src/ATen/native/MaxPooling.cpp [hunk 1]
```diff
@@ -58,33 +56,57 @@ Tensor max_pool1d_impl(
...
-  const int64_t NC = self.size(-2);
-  const int64_t IW = self.size(-1);
-  const int64_t KW = kernel_size[0];
-  const int64_t SJ = stride[0];
-  const int64_t PJ = padding[0];
-  const int64_t DJ = dilation[0];
-
   TORCH_CHECK(
-      KW > 0,
+      kernel_size[0] > 0,
```
[the remark this answers is not in the document]
Comment (deandyu) [reply aten/src/ATen/native/MaxPooling.cpp:60]: I did not look into the code for `max_pool2d` as the issue was focused on inconsistencies in `max_pool1d` and these inconsistencies were well documented in `pytorch/torch/testing/_internal/common_methods_invocations.py`

File: aten/src/ATen/native/MaxPooling.cpp [hunk 2]
```diff
@@ -126,9 +148,11 @@ Tensor max_pool1d(
       !self.device().is_cpu() ||
       isTensorSubclassLike(self)) {
     // Needs indices for grad and with_indices defines CUDA dispatch
+    check_max_pool1d(self, kernel_size, stride, padding, dilation);
```
Comment (ngimel) [inline aten/src/ATen/native/MaxPooling.cpp:151]: bring `check_max_pool1d` to above the conditional, as you are calling ti in all cases.

**Comment by deandyu** [comment]
Sorry if I've bothered you with notifications, I tried to click the re-request review button and it behaved weirdly on my end.

File: aten/src/ATen/native/MaxPooling.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (mruberry) [reply aten/src/ATen/native/MaxPooling.cpp:61]: OK

**Comment by mruberry** [comment]
Tests are running now, @deandyu. Once they run and reveal no issues relevant to this PR I think we'll be good to go. I'll just make a final check then.

**Comment by deandyu** [comment]
Hey @mruberry, have you had a chance to look at the tests yet? As far as I can tell, [(xla, 1, 1, linux.2xlarge)](https://github.com/pytorch/pytorch/actions/runs/3625578888/jobs/6149188685#logs) is the only test that fails, due to an issue that has been fixed in PR #90229.

**Review by mruberry — APPROVED** [review]
LGTM! Thanks @deandyu!

**Comment by deandyu** [comment]
@mruberry Could you approve the remaining workflow? Thanks!

**Pull request closed without merging**